File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.js
/**
 * @param {number[]} candies
 * @param {number} extraCandies
 * @return {boolean[]}
 */
var kidsWithCandies = function(candies, extraCandies) {
    const greatest =  Math.max(...candies);
    var kids = [];
    candies.forEach(candie =>{
        if(candie + extraCandies >=  greatest)
            kids.push(true);
        else
            kids.push(false);      
    })
    return kids;
};

File: 1431-kids-with-the-greatest-number-of-candies/1431-kids-with-the-greatest-number-of-candies.cpp
class Solution {
public:
    vector<bool> kidsWithCandies(vector<int>& candies, int extraCandies) {
        int greatest = 0;
        vector<bool> kids;
        for(const auto &candie :candies)
            if (candie > greatest)
                greatest = candie;
        
        for(const auto &candie :candies) {
            if (candie + extraCandies >= greatest)
                kids.push_back(true);
            else
                 kids.push_back(false);
        }
        return kids;
    }
};